cmake/main/CMakeLists.txt

```diff
@@ -0,0 +1,247 @@
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/RoutingLogic.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/SimpleNodeInfo.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/SimpleNodeInfoMockUserFile.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/SimpleStack.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/TcpDefs.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/TractionCvSpace.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/TractionDefs.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/TractionProxy.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/TractionTestTrain.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/TractionThrottle.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/TractionTrain.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/Velocity.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb/WriteHelper.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/os/os.c
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/os/stack_malloc.c
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/os/watchdog.c
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/Buffer.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/CanIf.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/ConfigUpdateListener.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/constants.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/Crc.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/errno_exit.c
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/FileUtils.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/format_utils.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/GcStreamParser.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/GcTcpHub.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/gc_format.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/GridConnect.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/GridConnectHub.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/HubDevice.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/HubDeviceSelect.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/ieeehalfprecision.c
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/JSHubPort.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/logging.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/Queue.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/SimpleQueue.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/SocketClient.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/socket_listener.cpp
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/utils/StringPrintf.cpp
+)
+
+
+
+set(DCCPP_SRCS
+  ${ATANI_DCC_BASEDIR}/src/ConfigurationManager.cpp
+  ${ATANI_DCC_BASEDIR}/src/DCCppESP32.cpp
+  ${ATANI_DCC_BASEDIR}/src/InfoScreen.cpp
+  ${ATANI_DCC_BASEDIR}/src/DCC/DCCProgrammer.cpp
+  ${ATANI_DCC_BASEDIR}/src/DCC/DCCSignalGenerator.cpp
+  ${ATANI_DCC_BASEDIR}/src/DCC/DCCSignalGenerator_RMT.cpp
+  ${ATANI_DCC_BASEDIR}/src/DCC/DCCSignalGenerator_Timer.cpp
+  ${ATANI_DCC_BASEDIR}/src/DCC/MotorBoard.cpp
+  ${ATANI_DCC_BASEDIR}/src/DCC/Turnouts.cpp
```

Comment:
Instead of one big src list, move to nested CMakeFile.txt in src, src/DCC, etc with includes or submodules.

Reply:
I will look into that. Currently I tried to keep all the cmake related files within the cmake subdirectory, but it certainly is more natural to have them in the package/project subdirectories